src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -87,6 +87,40 @@ private bool ProcessUserDefinitions(out UserDefinitionResult userDefinitionResul
             return true;
         }
 
+        /// <summary>
+        /// Process user script and returns user defined functions and named formulas.
+        /// </summary>
+        /// <param name="script">User script containing UDFs and/or named formulas.</param>
+        /// <param name="parseCulture">CultureInfo to parse the script.</param>
+        /// <returns>Tuple.</returns>
+        /// <exception cref="InvalidOperationException">Throw if the user script contains errors.</exception>
+        public static UserDefinitionResult Process(string script, CultureInfo parseCulture)
+        {
+            var options = new ParserOptions()
+            {
+                AllowsSideEffects = false,
+                Culture = parseCulture ?? CultureInfo.InvariantCulture
+            };
+
+            var sb = new StringBuilder();
+
+            UserDefinitions.ProcessUserDefinitions(script, options, out var userDefinitionResult);
```

Comment:
>ProcessUserDefinitions [](http://example.com/codeflow?start=28&length=22)

`ProcessUserDefinitions` also takes a `Features` argument, it may be worth passing one as well.